internal/provider/resource_fmc_device_ha_pair.go

```diff
@@ -0,0 +1,418 @@
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("API task for the new device failed: %s, %s", task.Get("message"), task.Get("description")))
+			return
+		}
+		if stat != "PENDING" && stat != "RUNNING" {
+			break
+		}
+		time.Sleep(atom)
+	}
+
+	check, err := r.client.Get(plan.getPath())
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to read object (GET), got error: %s, %s", err, check))
+		return
+	}
+	name := "items.#(name==" + url.QueryEscape(plan.Name.ValueString()) + ").id"
+	id := check.Get(name).String()
+	plan.Id = types.StringValue(id)
+	if plan.Id.ValueString() == "" {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("No device named %q: %s", plan.Name.ValueString(), check))
+		return
+	}
+
+	// Ending code to poll object
+	// plan.Id = types.StringValue(res.Get("id").String())
+	plan.fromBodyUnknowns(ctx, res)
+
+	tflog.Debug(ctx, fmt.Sprintf("%s: Create finished successfully", plan.Id.ValueString()))
+
+	diags = resp.State.Set(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	helpers.SetFlagImporting(ctx, false, resp.Private, &resp.Diagnostics)
+}
+
+// Section below is generated&owned by "gen/generator.go". //template:begin read
+
+func (r *DeviceHAPairResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state DeviceHAPair
+
+	// Read state
+	diags := req.State.Get(ctx, &state)
+	if resp.Diagnostics.Append(diags...); resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set request domain if provided
+	reqMods := [](func(*fmc.Req)){}
+	if !state.Domain.IsNull() && state.Domain.ValueString() != "" {
+		reqMods = append(reqMods, fmc.DomainName(state.Domain.ValueString()))
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", state.Id.String()))
+
+	urlPath := state.getPath() + "/" + url.QueryEscape(state.Id.ValueString())
+	res, err := r.client.Get(urlPath, reqMods...)
+
+	if err != nil && strings.Contains(err.Error(), "StatusCode 404") {
+		resp.State.RemoveResource(ctx)
+		return
+	} else if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to retrieve object (GET), got error: %s, %s", err, res.String()))
+		return
+	}
+
+	imp, diags := helpers.IsFlagImporting(ctx, req)
+	if resp.Diagnostics.Append(diags...); resp.Diagnostics.HasError() {
+		return
+	}
+
+	// After `terraform import` we switch to a full read.
+	if imp {
+		state.fromBody(ctx, res)
+	} else {
+		state.fromBodyPartial(ctx, res)
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("%s: Read finished successfully", state.Id.ValueString()))
+
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	helpers.SetFlagImporting(ctx, false, resp.Private, &resp.Diagnostics)
+}
+
+// End of section. //template:end read
+
+// Section below is generated&owned by "gen/generator.go". //template:begin update
+
+func (r *DeviceHAPairResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
after sending JSON it fails:

│ Failed to configure object (PUT), got error: HTTP Request failed: StatusCode 400, {"error":{"category":"FRAMEWORK","messages":[{"description":"FTD HA
│ container id is missing. Please refer the example JSON for the input attributes."}],"severity":"ERROR"}}

JSON sent:
```
{
  "id": "2ec141d8-b3d8-11ef-8c75-21eeb0cb0b9b",
  "name": "MyHADeviceName1",
  "type": "DeviceHAPair",
  "primary": {
    "id": "a6126182-b387-11ef-a1f3-a8ab0bf56ce8"
  },
  "secondary": {
    "id": "9f357070-b387-11ef-bf71-f1820f81db1b"
  },
  "ftdHABootstrap": {
    "isEncryptionEnabled": false,
    "useSameLinkForFailovers": true,
    "lanFailover": {
      "standbyIP": "192.168.0.2",
      "activeIP": "192.168.0.1",
      "logicalName": "failover",
      "subnetMask": "255.255.255.252",
      "useIPv6Address": false,
      "interfaceObject": {
        "name": "GigabitEthernet0/0",
        "id": "005056B0-9230-0ed3-0000-004294980595",
        "type": "PhysicalInterface"
      }
    }
  }
}
```

Reply:
Fixed it